lib/ansible/modules/storage/glusterfs/gluster_peer.py

```diff
@@ -0,0 +1,169 @@
+            removed from the pool. If the state is present, nodes will be
+            attached to the pool. If state is absent, nodes will be detached
+            from the pool.
+       required: true
+    nodes:
+       description:
+          - List of nodes that have to be probed into the pool.
+       required: true
+    force:
+       type: bool
+       default: "false"
+       description:
+          - Applicable only while removing the nodes from the pool. gluster
+            will refuse to detach a node from the pool if any one of the node
+            is down, in such cases force can be used.
+requirements:
+  - GlusterFS > 3.2
+notes:
+  - This module does not support check mode.
+'''
+
+EXAMPLES = '''
+- name: Create a trusted storage pool
+  gluster_pool:
+        state: present
+        nodes:
+             - 10.0.1.5
+             - 10.0.1.10
+
+- name: Delete a node from the trusted storage pool
+  gluster_pool:
+         state: absent
+         nodes:
+              - 10.0.1.10
+
+- name: Delete a node from the trusted storage pool by force
+  gluster_pool:
+         state: absent
+         nodes:
+              - 10.0.0.1
+         force: true
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from distutils.version import LooseVersion
+
+
+class Peer(object):
+    def __init__(self, module):
+        self.module = module
+        self.state = self.module.params['state']
+
+    def gluster_peer_ops(self):
+        nodes = self.module.params['nodes']
+        if not nodes:
+            self.module.fail_json(msg="nodes list cannot be empty")
+        force = 'force' if self.module.params.get('force') else ''
+        if self.state == 'present':
+            nodes = self.get_to_be_probed_hosts(nodes)
+            action = 'probe'
+            # In case of peer probe, we do not need `force'
+            force = ''
+        else:
+            action = 'detach'
+        cmd = []
+        for node in nodes:
+            cmd.append(' peer ' + action + ' ' + node + ' ' + force)
+        return cmd
+
+    def get_to_be_probed_hosts(self, hosts):
+        rc, output, err = self.module.run_command("gluster pool list")
+        peers_in_cluster = [line.split('\t')[1].strip() for
+                            line in filter(None, output.split('\n')[1:])]
+        try:
+            peers_in_cluster.remove('localhost')
+        except ValueError:
+            # It is ok not to have localhost in list
+            pass
+        hosts_to_be_probed = [host for host in hosts if host not in
+                              peers_in_cluster]
+        return hosts_to_be_probed
+
+    def call_peer_commands(self, cmds):
+        errors = []
+        for cmd in cmds:
+            rc, out, err = self._run_command('gluster', cmd)
+            if rc:
+                errors.append(err)
+        return errors
+
+    def get_output(self, errors):
+        if not errors:
+            self.module.exit_json(rc=0, changed=True)
```

Comment:
Note that you always return `changed=True` even when maybe something didn't change. You should document that.

Reply:
Am not talking about `check_mode`, but about `change=True`  / `changed=False`.

Reply:
I mean: if there is no way to detect if something have been changed, this module can behave like `command`/`shell` modules: `check_mode` not supported, almost (due to `creates`/`removes` parameters) always returning `changed=True`.

`gluster peer probe` command doesn't return an error if peer is already in list but a message is printed:
```
[root@proxy02 ~]# gluster peer probe 172.24.1.182
peer probe: success. Host 172.24.1.182 port 24007 already in peer list
[root@proxy02 ~]# echo $?
0
```
In order to avoid to report change when none occurs, the output could be parsed (should the LANG environment variable be set ?). Would not it be better to check output of `gluster pool list` or `gluster peer status` ?

Reply:
Well, by `check_mode` I personally understand that it doesn't support `--check` parameter passed to `ansible-playbook` command and not that the module will always return `changed=True`. Anyway I am OK with this module, but it doesn't really bring the Ansible power(mainly idempotency), as user can use `gluster` command with `command` module to have same results.